src/daft-dashboard/src/lib.rs

```diff
@@ -90,7 +127,253 @@ async fn deserialize<T: for<'de> Deserialize<'de>>(req: Req) -> ServerResult<Req
+    if row >= record_batch.len() || col >= record_batch.num_columns() {
+        return Err((
+            StatusCode::BAD_REQUEST,
+            anyhow::anyhow!("Row or column index out of bounds"),
+        ));
+    }
+
+    let column = record_batch.get_column(col);
+    let cell_html = daft_recordbatch::html_value(column, row, false);
+
+    let response = CellResponse {
+        value: cell_html,
+        data_type: format!("{:?}", column.data_type()),
+    };
+
+    Ok(response::with_body(StatusCode::OK, response))
+}
+
+fn generate_interactive_html(
+    record_batch: &RecordBatch,
+    df_id: &str,
+    host: &str,
+    port: u16,
+) -> String {
+    // Start with the basic table HTML from repr_html
+    let table_html = record_batch.repr_html();
+    // Build the complete interactive HTML with side pane layout
+    let mut html = vec![r#"
+        <style>
+        .dashboard-container {
+            display: flex;
+            gap: 20px;
+            max-width: 100%;
+            height: 100%;
+        }
+        .table-container {
+            flex: 1;
+            overflow: auto;
+        }
+        .side-pane {
+            width: 35%;
+            max-height: 500px;
+            border: 1px solid;
+            border-radius: 4px;
+            padding: 15px;
+            display: none;
+            overflow: auto;
+        }
+        .side-pane.visible {
+            display: flex;
+            flex-direction: column;
+        }
+        .side-pane-header {
+            display: flex;
+            justify-content: space-between;
+            align-items: center;
+            margin-bottom: 10px;
+            padding-bottom: 10px;
+            border-bottom: 1px solid;
+        }
+        .side-pane-title {
+            font-weight: bold;
+        }
+        .close-button {
+            cursor: pointer;
+        }
+        .side-pane-content {
+            word-wrap: break-word;
+            overflow: auto;
+        }
+        .dataframe td.clickable {
+            cursor: pointer;
+            transition: background-color 0.2s;
+        }
+        .dataframe td.clickable:hover {
+            opacity: 0.8;
+        }
+        .dataframe td.clickable.selected {
+            opacity: 0.6;
+        }
+        </style>
+        <div class="dashboard-container">
+            <div class="table-container">
+        "#
+    .to_string()];
+
+    // Add the table HTML with ID
+    html.push(format!(
+        r#"<div id="dataframe-{}">{}</div>"#,
+        df_id, table_html
+    ));
+
+    // Add the side pane
+    html.push(format!(
+        r#"
+            </div>
+            <div class="side-pane" id="side-pane-{df_id}">
+                <div class="side-pane-header">
+                    <div class="side-pane-title" id="side-pane-title-{df_id}">Cell Details</div>
+                    <button class="close-button" id="close-button-{df_id}">×</button>
```

Comment:
If we could set a background color on the close button and have the background color change on hover that would be nice! Optional though